backend/src/main/java/com/woowacourse/thankoo/coupon/domain/Coupon.java

```diff
@@ -113,6 +115,39 @@ public boolean isSameCouponContent(final CouponContent couponContent) {
         return this.couponContent.equals(couponContent);
     }
 
+    public void complete(final Long memberId) {
+        validateExchangedMember(memberId);
+        validateStatus();
+        publishCouponCompleteEvent();
+        couponStatus = CouponStatus.USED;
+    }
+
+    private void validateExchangedMember(final Long memberId) {
+        if (!isExchangedMember(memberId)) {
+            throw new InvalidMemberException(ErrorType.CAN_NOT_COMPLETE_MISMATCH_MEMBER);
+        }
+    }
+
+    private boolean isExchangedMember(final Long memberId) {
+        return isSender(memberId) || isReceiver(memberId);
+    }
+
+    private void validateStatus() {
+        if (!isCompleteStatus()) {
+            throw new InvalidCouponException(ErrorType.CAN_NOT_COMPLETE);
+        }
+    }
+
+    private boolean isCompleteStatus() {
+        return !couponStatus.isReserved() && !couponStatus.isUsed() && !couponStatus.isExpired();
+    }
```

Comment:
```suggestion
    private void validateStatus() {
        if (!couponStatus.canChangeComplete()) {
            throw new InvalidCouponException(ErrorType.CAN_NOT_COMPLETE);
        }
```
이 부분은 CouponStatus에서 해주는건 어떨까요??

Reply:
status에게 던지도록 수정했어요!